opensquirrel/circuit.py

```diff
@@ -166,3 +149,31 @@ def export(self, fmt: ExportFormat | None = None) -> Any:
             return cqasmv1_exporter.export(self)
         msg = "unknown exporter format"
         raise ValueError(msg)
+
+
+@dataclass(init=False)
+class PhaseMap(Circuit):
+
+    def __init__(self, phase_map: ArrayLike[np.complex128]):
+        """Initialize a PhaseMap object."""
+        self.phase_map = phase_map
+
+    def __contains__(self, qubit: QubitLike) -> bool:
+        """Checks if qubit is in the phase map."""
+        if qubit in self.phase_map:
+            return True
+
+        return False
+
+    def set_phase_map(self, phase_map: NDArray[np.complex128]) -> None:
+        self.qubit_phase_map = phase_map
+
```

Comment:
```suggestion
```

I think this is a copy construction, and should be taken care of in the/a constructor.